### DISCUSSION TITLE
How to "Define the TESSDATA_PREFIX environment variable to point to your specific folder "

### DISCUSSION THREAD

**Discussion by al808909** [Audiveris/audiveris#740] [Q&A] [answered]
Would someone be willing to explain the steps involved for a mac m1 running macOs 14.5.
Audiveris is working, but can't find the tessdata folder that I conveniently put in the audiveris folder :-)
Thanks Alex.

**Comment by hbitteur** [comment]
A few words of explanation.

Audiveris delegates the OCR task to a Tesseract library.
This Tesseract library knows nothing about Audiveris and its folders.
It relies on the system environment variable TESSDATA_PREFIX to find where the languages files are located.

So, my advice would be to set TESSDATA_PREFIX so that it points to your tessdata folder (which may be within your Audiveris folder, BTW).

**Comment by al808909** [comment, marked as answer]
FOR MAC:

I got this working on a mac mini m1 running 14.5:

Open the terminal app, type:

echo "TESSDATA_PREFIX=where you put the tessdata folder">> ~/.zshrc

In my case I put the tessdata folder in the config folder in the audiveris folder that is in my user directory:

echo "TESSDATA_PREFIX=/Users/al808909/audiveris/config/tessdata" >> ~/.zshrc

to save the "environment variable" you then need to type:

source ~/.zshrc

That's it I believe, you can check it all worked by typing:

env | grep "TESSDATA_PREFIX"

which returns in my case:

TESSDATA_PREFIX=/Users/al808909/audiveris/config/tessdata

The youtube video that helped me (at 1:52) was:

https://www.youtube.com/watch?v=dl_jgYr0rxU&t=174s

FOR WINDOWS:

Go to the start menu, type in the search field "environment variable" to get the environment variable control panel to appear.
Open it, click Environment Variables, at the bottom click on New, enter TESSDATA_PREFIX for Variable name and paste-in the path of where-you-put-the-tessdata folder for Variable Value, click ok a couple of times and then Apply, and you're done.

The youtube video that helped me (at 7:29) was:

https://www.youtube.com/watch?v=AWkqC6VFIyQ&t=467s

Hope this is useful.
